test/dnsregistrar/TestOffchainDNSResolver.js

```diff
@@ -0,0 +1,206 @@
+
+    registrar = await DNSRegistrarContract.new(
+      ZERO_ADDRESS, // Previous registrar
+      offchainResolver.address,
+      dnssec.address,
+      suffixes.address,
+      ens.address
+    )
+    await root.setController(registrar.address, true)
+    await root.setController(accounts[0], true)
+  })
+
+  it('should respond to resolution requests with a CCIP read request to the DNS gateway', async function() {
+    const pr = await PublicResolver.at(offchainResolver.address);
+    const DNSGatewayInterface = new ethers.utils.Interface(IDNSGateway.abi);
+    const dnsName = utils.hexEncodeName('test.test');
+    const callData = pr.contract.methods['addr(bytes32)'](namehash.hash('test.test')).encodeABI();
+    await expect(offchainResolver.resolve(
+      dnsName,
+      callData
+    )).to.be.revertedWith('OffchainLookup('
+      + '"' + offchainResolver.address + '", '
+      + '["https://localhost:8000/query"], '
+      + '"' + DNSGatewayInterface.encodeFunctionData('resolve', [dnsName, 16]) + '", '
+      + '"' + ethers.utils.id('resolveCallback(bytes,bytes)').slice(0, 10) + '", '
+      + '"' + ethers.utils.defaultAbiCoder.encode(['bytes', 'bytes'], [dnsName, callData]) + '"'
+      + ')'
+    );
+  })
+
+  function doResolveCallback(name, texts, callData) {
+    const proof = [
+      hexEncodeSignedSet(rootKeys(expiration, inception)),
+      hexEncodeSignedSet(testRrset(name, texts)),
+    ];
+    const response = ethers.utils.defaultAbiCoder.encode(
+      ["tuple(bytes, bytes)[]"],
+      [proof]
+    );
+    const dnsName = utils.hexEncodeName(name);
+    const extraData = ethers.utils.defaultAbiCoder.encode(
+      ["bytes", "bytes"],
+      [dnsName, callData]
+    );
+    return offchainResolver.resolveCallback(response, extraData);
+  }
+
+  it('handles calls to resolveCallback() with valid DNS TXT records containing an address', async function() {
+    const name = "test.test";
+    const testAddress = '0xfefeFEFeFEFEFEFEFeFefefefefeFEfEfefefEfe';
+    await ownedResolver.setAddr(namehash.hash(name), testAddress);
+    const pr = await PublicResolver.at(offchainResolver.address);
+    const callData = pr.contract.methods['addr(bytes32)'](namehash.hash(name)).encodeABI();
+    const result = await doResolveCallback(name, [`ENS1 ${ownedResolver.address}`], callData);
+    expect(ethers.utils.defaultAbiCoder.decode(['address'], result)[0]).to.equal(testAddress);
+  })
+
+  it('handles calls to resolveCallback() with extra data and a legacy resolver', async function() {
+    const name = "test.test";
+    const testAddress = '0xfefeFEFeFEFEFEFEFeFefefefefeFEfEfefefEfe';
+    await ownedResolver.setAddr(namehash.hash(name), testAddress);
+    const pr = await PublicResolver.at(offchainResolver.address);
+    const callData = pr.contract.methods['addr(bytes32)'](namehash.hash(name)).encodeABI();
+    const result = await doResolveCallback(name, [`ENS1 ${ownedResolver.address} blah`], callData);
```

Comment:
what does the `blah` suppose to do? Is this just testing that the callback returns the same result with or without the `blah` text without throwing an error?

Reply:
That's right.